pkg/asset/installconfig/alibabacloud/client.go

```diff
@@ -0,0 +1,127 @@
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/auth"
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/auth/credentials"
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/auth/credentials/provider"
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/responses"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ecs"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/resourcemanager"
+)
+
+// Environmental virables
+const (
+	ENVAccessKeyID     = "ALIBABA_CLOUD_ACCESS_KEY_ID"
+	ENVAccessKeySecret = "ALIBABA_CLOUD_ACCESS_KEY_SECRET"
+)
+
+// Client makes calls to the Alibaba Cloud API.
+type Client struct {
+	sdk.Client
+	credential auth.Credential
+}
+
+func NewClientWithOptions(regionId string, config *sdk.Config, credential auth.Credential) (client *Client, err error) {
+	client = &Client{}
+	err = client.InitWithOptions(regionId, config, credential)
+	return
+}
+
+func NewClient(regionId string) (client *Client, err error) {
+	credential, err := getCredentials()
+	if err != nil {
+		return nil, err
+	}
+
+	config := sdk.NewConfig()
+
+	client, err = NewClientWithOptions(regionId, config, credential)
+	client.credential = credential
+	return
+}
+
+func getCredentials() (credential auth.Credential, err error) {
+	// Get AccessKey and AccessKeySecret information from the enviroment
+	// usage: https://github.com/aliyun/alibaba-cloud-sdk-go/blob/7259de46d58ef905c66e04babf791190512a85da/docs/2-Client-EN.md#1-environment-credentials
+	credential, err = provider.NewEnvProvider().Resolve()
+	if credential != nil {
+		return credential, nil
+	}
+
+	// Get AccessKey and AccessKeySecret information from configuration file,default path:"~/.alibabacloud/credentials"
+	// usage: https://github.com/aliyun/alibaba-cloud-sdk-go/blob/7259de46d58ef905c66e04babf791190512a85da/docs/2-Client-EN.md#2-credentials-file
+	credential, err = provider.NewProfileProvider().Resolve()
+	if credential != nil {
+		return credential, nil
+	}
+
+	// Get AccessKey and AccessKeySecret information via interactive command line
+	credential, err = askCredentials()
+	if credential != nil {
+		return credential, nil
+	}
+
+	return nil, err
+}
+
+func askCredentials() (auth.Credential, error) {
+	var access_key_id string
+
+	err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Alibaba Cloud Access Key ID",
+				Help:    "The AccessKey ID is used to identify a user.\nhttps://www.alibabacloud.com/help/doc-detail/53045.html",
+			},
+		},
+	}, &access_key_id)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var access_key_secret string
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Password{
+				Message: "Alibaba Cloud Secret Access Key",
+				Help:    "The AccessKey secret is used to verify a user. You must keep your AccessKey secret strictly confidential.",
+			},
+		},
+	}, &access_key_secret)
+
+	if err != nil {
+		return nil, err
+	}
+
+	os.Setenv(ENVAccessKeyID, access_key_id)
+	os.Setenv(ENVAccessKeySecret, access_key_secret)
```

Comment:
Thanks Patrick,
1.For commit history and messages, can i rebase these commits to squashing changes refer to the guidelines your point to.
2.We will using new Terraform stages framework as soon as possible
3.There is not cluster destroy code, we will add the code in a later PR

Reply:
Sounds good. In case my above comment was unclear, you do not need to squash all of your commits. It is fine to have multiple commits, in fact we prefer it as long as the commits are logical units, but please squash any changes that are added in one commit and changed in a subsequent one.

Reply:
Thanks Patrick, I reorganized the commit history according to your suggestion, maybe it will be helpful for review

Reply:
> Thanks Patrick, I reorganized the commit history according to your suggestion, maybe it will be helpful for review

Thank you this definitely looks better. There are still no commit messages and the titles are not descriptive enough. For example you have a commit with only the title:
```add the terraform templates```

I would suggest you add a platform title
```
Alibaba: Add Terraform Templates
```

Even better would be to add a short description under the title describing the work.

Reply:
> > Thanks Patrick, I reorganized the commit history according to your suggestion, maybe it will be helpful for review
> 
> Thank you this definitely looks better. There are still no commit messages and the titles are not descriptive enough. For example you have a commit with only the title:
> `add the terraform templates`
> 
> I would suggest you add a platform title
> 
> ```
> Alibaba: Add Terraform Templates
> ```
> 
> Even better would be to add a short description under the title describing the work. Please make sure you at least add a Platform to all appropriate commit messages so that we can git a sense of what the commit is when we run `git log`.

OK, I have add commit messages and a platform name